### ISSUE TITLE
component/unit tests

### ISSUE THREAD

**Issue by PhilKsr** [Frachtwerk/essencium-frontend#681]
**Description**
As we decided to only go with unit and e2e tests, the component tests should be removed. In addition we should check the unit tests for coverage and implement missing tests

**Scope**
- component tests
- unit tests

- [ ] remove component tests
- [ ] implement unit tests
- [ ] update docs

**Comment by lwydra** [comment]
As far as I have understood so far, this decision doesn't seem to be final, so I'll add my 2¢ to it:

I would advise against completely abandoning component/integration tests in favor of e2e tests, as the cost to (initially) write and (later on) maintain them would be far higher if we want to achieve a similar coverage as with a combination of unit, component and e2e tests.
In addition, e2e tests might not be as meaningful when components from the `lib`-package are used in a different context in downstream applications, e.g. when those applications use features from components that are not used on Essencium pages.

I had a few ideas on how the creation of component tests could be simplified so they pose less of an overhead:

- keep components as compact as possible and split them into sub-components where feasible: this is already the case for most components, so not much to do here. Besides easier to test, because smaller packages, this also lessens the mental load when working on the components itself
- centralize test data mocks in a common directory: this would move lengthy object definitions out of the actual test classes and makes them reusable for other components without having to duplicate them
- add [MSW](https://mswjs.io/docs/) to centrally mock backend endpoints: in case any components make any network requests, the respective endpoints could be centrally mocked here (either with static responses or responses based on the parameters sent). This way, not every test would need to mock `window.fetch` anew and we can also use a central set of test data (s. point above).

Tl;dr: If voting's still open, I cast my vote for e2e tests in `app` and component tests in `lib`, otherwise ignore 😆
